### ISSUE TITLE
Did the official demo use better model than vit_h model?

### ISSUE THREAD

**Issue by zxj7b80** [facebookresearch/segment-anything#347]
When I provide multiply points for the model sam_vit_h_4b8939.pth, bad result occurs. It's different from the official demo 'https://segment-anything.com/demo'. In official demo, better segment result occurs with more points prompt. So I wonder whether the official demo used better model than sam_vit_h_4b8939.pth.

**Comment by githublqs** [comment]
I also have this question

**Comment by githublqs** [comment]
I also have this question [https://github.com/facebookresearch/segment-anything/issues/346](url)

**Comment by AlexLee1235** [comment]
In my case, when I prompt it iteratively (add one point a time and use old mask as prompt), it's result is more like the web demo.

for i in range(1,len(samPoints)):
        masks, scores, logits = predictor.predict(
            point_coords = np.array(samPoints[0:i]),
            point_labels = np.array(pointFlag[0:i]),
            mask_input = prev[None, :, :] if i > 1 else None,
            multimask_output = False,
        )
        prev =  logits[np.argmax(scores), :, :]

**Comment by sunsunnyshine** [comment]
I'm confused too!!
The later is the demo result.
![image](https://github.com/facebookresearch/segment-anything/assets/87694659/abe9fdcf-1122-4679-8369-6de03b91c3dc)
![image](https://github.com/facebookresearch/segment-anything/assets/87694659/40cadde1-166b-4bb1-a5b8-31c407162f6e)

**Comment by sunsunnyshine** [comment]
I use the iterative prompt,and got this.
The result has improved slightly, but it still doesn't meet the requirements compared to the demo.
![image](https://github.com/facebookresearch/segment-anything/assets/87694659/6cf0f1a9-b51e-43d5-98f9-bf0c7b8aaefc)
It seems like less point bright better results??
![image](https://github.com/facebookresearch/segment-anything/assets/87694659/98918769-9f81-4aa6-8be4-53780816d290)

**Comment by sunsunnyshine** [comment]
> In my case, when I prompt it iteratively (add one point a time and use old mask as prompt), it's result is more like the web demo.
> 
> for i in range(1,len(samPoints)): masks, scores, logits = predictor.predict( point_coords = np.array(samPoints[0:i]), point_labels = np.array(pointFlag[0:i]), mask_input = prev[None, :, :] if i > 1 else None, multimask_output = False, ) prev = logits[np.argmax(scores), :, :]

Hello, may I ask how the result looks after this processing? If you could provide a visual representation, I would greatly appreciate it!

**Comment by 315386775** [comment]
following

**Comment by Gi-gigi** [comment]
@AlexLee1235 

> In my case, when I prompt it iteratively (add one point a time and use old mask as prompt), it's result is more like the web demo.
> 
> for i in range(1,len(samPoints)): masks, scores, logits = predictor.predict( point_coords = np.array(samPoints[0:i]), point_labels = np.array(pointFlag[0:i]), mask_input = prev[None, :, :] if i > 1 else None, multimask_output = False, ) prev = logits[np.argmax(scores), :, :]


Hi friend, thank you very much for your guidance and advice. However, I have a few questions I'd like to ask:

Are the samPoints something you customized?
If they are customized, how should I modify the code to iteratively feed the points into the model? I'd appreciate your guidance, and I would be very grateful if you could provide it.

**Comment by ssjhhj** [comment]
> 就我而言，当我迭代地提示它（一次添加一个点并使用旧掩码作为提示）时，它的结果更像是网络演示。
> 
> 对于范围内的 i（1，len（samPoints））： 掩码、分数、logits = predictor.predict（ point_coords = np.array（samPoints [0：i]）， point_labels = np.array（pointFlag [0：i]）， mask_input = prev [None，：，：]如果 i> 1否则为None， multimask_output = False， ） prev = logits [np.argmax（scores），：，：]

yyds，You're a genius.
